python/test/unit/language/test_core.py

```diff
@@ -696,46 +703,57 @@ def duplicate_dim2(dummy, N: tl.constexpr):
 
     N = 32
     dummy_tensor = torch.empty((), device=device)
+    error_class = ValueError if is_interpreter() else triton.CompilationError
```

Comment:
IMO either we should not test the error class at all, or it's a bug that the interpreter returns a different error class.  I tend to think it's a bug, because the error class is user-visible.

You don't have to change it in this PR, but it would be nice not to have to have these `if` statements everywhere that we check exceptions.

Reply:
> It's probably not a bug.

Yeah, that's why it happens.  But by "probably a bug" I mean, it's something we should probably fix, so it's consistent across backends.

Reply:
Comments added. I'll address this issue soon.